cartoframes/dataset.py

```diff
@@ -84,6 +87,66 @@ def from_geodataframe(cls, gdf):
     def from_geojson(cls, geojson):
         return cls(gdf=load_geojson(geojson), state=cls.STATE_LOCAL)
 
+    @property
```

Comment:
Adding a comment here about the `is_sync` attribute and properties, but it's a general comment.

Looks like we could generalize the implementation of these properties as they always do the same stuff: several checks on the property + maintaining the `is_sync` attribute.

In carto-python we used Python metaclasses (via the [pyrestcli library](https://github.com/danicarrion/pyrestcli/blob/master/pyrestcli/resources.py#L67-L92)) which allowed to specify a [number of fields in a class](https://github.com/CartoDB/carto-python/blob/master/carto/datasets.py#L39-L82) and do operations on them.

I'm not a super-expert on implementing this kind of stuff in python, but I'd suggest to do some research to see how we could improve this (could be metaclasses, something custom or anything else). I'm sure there are frameworks in Python (like Django or others) that have already faced this kind of problem (not saying to use them, but take a look to possible implementations).